src/pages/dogs/[id]/index.jsx

```diff
@@ -108,14 +112,25 @@ export default function IndividualDogPage() {
       }
     );
   }
-  const { data: session } = useSession();
+  const { data: session, status } = useSession();
   const user = session?.user;
 
+  useEffect(() => {
+    if (status === "authenticated") {
```

Comment:
We shouldn't have to check status anymore, since that is handled by the `SessionManager`. We know that the SessionManager always gives us a not-null, authenticated session. I don't think we even have to wrap this in a useEffect. Just call `fetch("/api/users/{session.user._id})`